File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0
    let min = prices[0]
    for(let i = 1; i< prices.length; i++){
        let current = 0
        if(prices[i] < min) min = prices[i]
        else{
            current = prices[i] - min
        }
        if(current > maxProfit) maxProfit = current
    }
    return maxProfit
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit = 0
        minpri = prices[0]
        for i in range(1, len(prices)):
            if(prices[i] <minpri): 
                minpri = prices[i]
                continue
            current = prices[i] - minpri
            if(current > maxprofit): maxprofit = current
        return maxprofit
            